src/utils/info/pools/index.ts

```diff
@@ -24,19 +24,161 @@ import { MercuryPair } from "../../../../pages/api/pairs";
 import { MercuryEvent } from "../../../../pages/api/events";
 import { adjustAmountByDecimals } from "utils/utils";
 import { fillChart } from "utils/complete-chart";
-
 export const stellarNetworkDict = {
   MAINNET: Networks.PUBLIC,
   TESTNET: Networks.TESTNET,
 };
 
-export const buildPoolsInfo = async (
+export const buildPoolsInfoV2 = async (
   data: MercuryPair[],
   tokenList: TokenType[],
   network: Network
 ) => {
+  const filteredData = data.filter((pool) => {
+    const tokenAExists = tokenList.some((token) => token.contract === pool.tokenA);
+    const tokenBExists = tokenList.some((token) => token.contract === pool.tokenB);
+    return tokenAExists && tokenBExists;
+  });
+
   const sdkNetwork = stellarNetworkDict[network];
+  const router = getRouterFromPools(filteredData, sdkNetwork);
 
+  // const XLM = tokenList.find((token: TokenType) => token.contract === "XLM");
+  // if (!XLM) throw new Error("XLM token not found in the token list.");
+  const USDC = tokenList.find((token) => token.code === "USDC");
+  if (!USDC) throw new Error("USDC token not found in the token list.");
+  const USDx = tokenList.find((token: TokenType) => token.code === "USDx");
+  if (!USDx) throw new Error("USDx token not found in the token list.");
+
+  const forceGetTokenPrice = true; // FALSE TO SET STATIC PRICES HARDCODE (XLM/USDC/USDx)
+  const staticTokenPrices: Record<string, number> = {
+    // [XLM.contract]: 0.529266,
```

Comment:
???

Reply:
I explained it earlier, If the boolean is true the flow is gonna take prices from router...